lib/components/Atom/WheelPicker/time-wheel-picker/index.tsx

```diff
@@ -0,0 +1,321 @@
+            value: minuteValue.value || inputSelectedDate.get("minute"),
+        });
+
+        const minutesData = [];
+        for (let i = 0; i < 60; i += minutesInterval) {
+            minutesData.push({
+                label: `0${i}`.slice(-2),
+                value: i,
+            });
+        }
+        setMinutes([...minutesData]);
+    }, []);
+
+    useEffect(() => {
+        if (!is12Hour && startTimeIn24Format) {
+            const minutesData = [];
+
+            let startMinute = 0;
+            if (inputStartDate.get("hour") === hourValue.value) {
+                startMinute =
+                    inputStartDate.get("minute") +
+                    (inputStartDate.get("minute") % minutesInterval
+                        ? minutesInterval -
+                          (inputStartDate.get("minute") % minutesInterval)
+                        : 0);
+                if (startMinute === 60) {
+                    setHourValue({
+                        label: `0${hourValue.value + 1}`.slice(-2),
+                        value: hourValue.value + 1,
+                    });
+                    setHours(
+                        hours?.filter((hour) => hour.value > hourValue.value),
+                    );
+                }
+            }
+
+            for (
+                let i = startMinute;
+                i <= (inputEndDate.get("minute") || 59);
+                i += minutesInterval
+            ) {
+                minutesData.push({
+                    label: `0${i}`.slice(-2),
+                    value: i,
+                });
+            }
+
+            setMinutesArray(minutesData);
+        }
+    }, [hourValue]);
+
+    useEffect(() => {
+        if (setSelectedValue) {
+            setSelectedValue(
+                (hourValue.label +
+                    ":" +
+                    minuteValue.label +
+                    (is12Hour ? " " + meridiemValue.label : "")) as string,
+            );
+        }
+
+        if (setSelectedTime)
+            setSelectedTime(
+                hourValue.value +
+                    ":" +
+                    minuteValue.value +
+                    (is12Hour ? " " + meridiemValue.value : ""),
+            );
+    }, [hourValue, minuteValue, meridiemValue]);
+
+    const handleKeyDown = (e: KeyboardEvent<Element>, index: number) => {
+        setColRef(new Array(is12Hour ? 3 : 2).fill(false));
+
+        switch (e.key) {
+            case KEY.ARROW_RIGHT:
+                if (index === 0) {
+                    setColRef([false, true, false]);
+                } else if (index === 1) {
+                    setColRef([false, false, true]);
+                }
+
+                break;
+            case KEY.ARROW_LEFT:
+                if (index === 2) {
+                    setColRef([false, true, false]);
+                } else if (index === 1) {
+                    setColRef([true, false, false]);
+                }
```

Comment:
```suggestion
                setColRef([index === 1, index === 2, false]);
```

Reply:
that is an awesome idea, Thanks @maryia-deriv